typing/typecore.ml

```diff
@@ -690,11 +701,15 @@ type pattern_variable =
   }
 
 type module_variable =
-  string loc * Location.t
+  {
```

Comment:
The extra field is `tu_uid : Uid.t`. I thought it would be a bit odd to create two independent `Uid.t`s for a module pattern that occurs on both sides of an or-pattern, and then drop the latter one, so I preferred not creating the `Uid.t` until after the or-pattern is processed and we're ready to add the module variable to the environment. On reflection, this doesn't seem so bad (or at least is outweighed by the benefit of reducing the number of concepts) so I'll take the suggestion.